README.md

```diff
@@ -2,6 +2,324 @@
+of the injection rule. In this case, it means:
+
+1. For every new `component` created,
+2. Set its `store` property to
+3. The instance of `store:main`
+
+In other words, every time `container.lookup('component:profile')` gets called,
+something like this is happening under the hood:
+
+```js
+let store = container.lookup('store:main');
+return ProfileComponent.create({ store });
+```
+
+The nice thing about injections is that we can set up a rule once and not worry
+about the details of where and how instances actually get created. This
+separation of concerns allows for less brittle code.
+
+You've also now seen why specifiers contain information about both name and
+type. Injections let us specify rules that apply to all instances of a
+component, say, without having to repeat that rule for every component in the
+system.
+
+## Resolvers: Mapping to the File System
+
+So far, we've always had to tell the `Registry` about a class before we're able
+to get an instance from the `Container`. But if we're being good developers, and
+organizing our code well and being consistent in our naming, shouldn't our app
+be able to find our classes automatically?
+
+That's exactly what the `Resolver` helps us do. With a resolver, we can define
+rules that map specifiers (like `component:profile`) on to module names (like
+`app/components/profile.js`).
+
+A simple `Resolver` implements a single method, `retrieve()`, which takes a
+specifier and returns the associated class.
+
+Lets write a resolver that will load the component class using CommonJS instead of
+having to eagerly register every component in our app:
+
+```js
+class Resolver {
+  retrieve(specifier) {
+    let [type, name] = specifier.split(':');
+
+    if (type !== 'component') { throw new Error("Unsupported type"); }
+    return require(`./app/${type}s/${name}.js`);
+  }
+}
+
+let registry = new Registry();
+let resolver = new Resolver();
+let container2 = new Container(registry, resolver);
+
+// Make sure components aren't singletons
+registry.registerOption('component', 'singleton', false);
+
+// Requires and instantiates `./app/components/admin-page.js`:
+let adminPage = container2.lookup('component:admin-page');
+```
+
+Note that `retrieve()` *must* return synchronously. Your module loader therefore
+must return synchronously, as it does in this CommonJS example. If you're using an
+asynchronous module loader, you'll need to make sure modules are loaded before you
+start instantiating objects.
+
+As a general rule, this package is designed to be synchronous to achieve maximum
+performance; it is your responsibility to ensure that code is ready before it is
+needed.
+
+One last thing: you may have noticed that the container in this example has both
+a registry and a resolver. The container will look for classes in both, but the
+registry always takes precedence. If the registry is empty, the container will
+fall back to asking the resolver for its help.
```

Comment:
This is grammatically correct. It specifies that the help is coming from the resolver. In "asking the resolver for help" the possession is implicitly the resolver's, but here it's made explicit: "asking the resolver for (its) help."